docs/src/pages/components/data-grid/sorting/sorting.md

```diff
@@ -4,34 +4,92 @@ title: Data Grid - Sorting
+
+{{"demo": "pages/components/data-grid/sorting/BasicExampleDataGridPro.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Pass sorting rules to the grid
+
+### Structure of the model
+
+The sort model is a list of sorting items.
+Each item represents a sorting rule and is composed of several elements:
+
+- `sortingItem.field`: the field on which we want to apply the rule
+- `sortingItem.sort`: the direction of the sorting (`'asc'`, `'desc'`, `null` or `undefined`). If `null` or `undefined`, the rule will not be applied.
+
+### Initialize the sort model
+
+To initialize the sort model without controlling it, provide the model to the `initialState` prop.
+
+```jsx
+<DataGrid
+  initialState={{
+    sorting: {
+      sortModel: [{ field: 'rating', sort: 'desc' }],
+    },
+  }}
+/>
+```
+
+{{"demo": "pages/components/data-grid/sorting/InitialSort.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Controlled sort model
+
+Use the `sortModel` prop to control the sorting rules applied on the rows.
+
+You can use the `onSortModelChange` prop to listen to changes in the sorting rules and update the prop accordingly.
+
+{{"demo": "pages/components/data-grid/sorting/ControlledSort.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Disable the sorting
+
+By default, all columns are sortable.
+To disable the filter on a column, set the `sortable` property of `GridColDef` to `false`.
+In the example below, the _rating_ column can not be sorted.
+
+```tsx
+<Datagrid columns={[...columns, { field: 'rating', sortable: false }]} />
+```
+
+{{"demo": "pages/components/data-grid/sorting/DisableSortingGrid.js", "bg": "inline", "defaultCodeOpen": false}}
 
 ## Custom comparator
 
-The grid handles sorting and applies different comparators in columns according to their types.
-The component handles sorting natively for the following types:
+A comparator determines how two cell values should be sorted.
 
-- string
-- number
-- date
-- dateTime
+Each column type comes with a default comparator method.
+You can re-use them by importing the following functions:
+
+- `gridStringNumberComparator` (used by the `string` and `singleSelect` columns)
+- `gridNumberComparator` (used by the `number` and `boolean` columns)
+- `gridDateComparator` (used by the `date` and `date-time` columns)
 
 To extend or modify this behavior in a specific column, you can pass in a custom comparator, and override the `sortComparator` property of the `GridColDef` interface.
 
-In the example below, the `username` column combines `name` and `age`, but it is sorted by `age` using a custom comparator:
+In the example below:
+
+- the "Name" column combines the `name` and `isAdmin` fields. The sorting is based on `isAdmin` and then on `name`, if necessary. It re-uses the built-in sorting comparator.
+- the "Created on" column sorting is based on the day of the month of the `createdOn` field. It is a fully custom sorting comparator.
```

Comment:
I feel we should break this into **two** examples.
Perhaps the first showing only the fully custom comparator, and then a second showing how to extend it using the built-in methods.